Allow customizing domain in deploydocs (#2457)
JuliaDocs/Documenter.jl: docs/src/man/hosting.md

https://USER_NAME.github.io/PACKAGE_NAME.jl/dev
```

> !!! tip
>     A custom domain can be set up with the keyword argument `cname` to
>     [`deploydocs`](@ref), which will then be used as the base URL for the documentation.

Reviewer: ```suggestion
!!! tip
    If you need Documenter to maintain [a `CNAME` file](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site) for you can use the `cname` argument of [`deploydocs`](@ref) to specify the domain.
```
